sphinx_changelog/towncrier.py

```diff
@@ -15,6 +15,25 @@
 from towncrier._settings import load_config_from_options
 
 
+class _Config:
+    """
+    Wrapper class to make the (private) config object in towncrier
+    subscriptable.
+
+    In version 22.12.0 the config object was turned
+    from a dict to a dataclass, so this can be removed if/when
+    towncrier is pinned to >= 22.12.
```

Comment:
I have been thinking about towncrier releases and this package.

My thinking is that we should hard pin a single release of towncrier for this package and update it explicitly every time. My reasoning for this unusual decision is two fold:

1) We are relying on a private unstable API of towncrier, which they make no promises about and therefore break regularly.
1) This is not library package it's a sphinx extension, only needed for doc builds, which 99.99% of the time will have no other need for towncrier.

The only potential issue I can see is if towncrier release a change which breaks the config, packages will need to keep this and the towncrier versions in sync, but I don't think that should be an issue as long as we keep up with tc releases.

So, all that being said, I suggest you update the code to rely only on 22.12 and hard pin it in setup.cfg.